### PR TITLE
Update dependencies to resolve vulnerabilities

### PR DESCRIPTION
**Pull request by chriswilty**
This is a general sweep of the package dependencies, as there were some critical and high vulnerabilities, as well as some deprecated packages (such as document-register-element and the unscoped babel libs).

Alongside these, there are some lightweight formatting changes due to SASS deprecation warnings and prettier upgrade to v3 (which is now even more opinionated 😉).

This is a BREAKING CHANGE: Node 18+ is required by rollup v4 and semantic-release v22

### CODE REVIEW

File: src/applause-button.js [hunk 1]
```diff
@@ -1,33 +1,27 @@
-import "document-register-element/build/document-register-element";
+import "@ungap/custom-elements";
 
 const VERSION = "3.3.0";
 const API = "https://api.applause-button.com";
 
 const getClaps = (api, url) =>
-  // TODO: polyfill for IE (not edge)
-  fetch(`${api}/get-claps` + (url ? `?url=${url}` : ""), {
-    headers: {
```
Comment (dogle-scottlogic) [inline src/applause-button.js:9]: I assume we no longer need the Content-Type header for the get request?

File: src/applause-button.js [hunk 2]
```diff
@@ -40,26 +34,18 @@ const toggleClass = (element, cls) => {
...
+  let timer = null;
+  return function () {
+    const context = this,
       args = arguments;
     clearTimeout(timer);
     timer = setTimeout(() => fn.apply(context, args), delay);
   };
 };
 
-// https://github.com/WebReflection/document-register-element#v1-caveat
```
Comment (dogle-scottlogic) [inline src/applause-button.js:52]: is this baked into the ungap import?

File: src/applause-button.test.js [hunk 3]
```diff
@@ -111,7 +113,7 @@ describe("Applause button", () => {
     it("send the updated clap count to the server", async () => {
       await createPage();
       await clickTheButton();
-      await pause(2500);
+      await pause(2200);
```
Comment (dogle-scottlogic) [inline src/applause-button.test.js:116]: For my understanding, why do we need this change?

**Review by dogle-scottlogic — APPROVED** [review]
Looks good to me - couple of questions but no issues I can see

**Comment by dogle-scottlogic** [comment]
can we close off the dependabot PRs after this goes in?

File: src/applause-button.js (hunk 1 above)
Comment (chriswilty) [reply src/applause-button.js:9]: Nope, cos we're not sending any content. Probably a copy-paste job from the POST request 😊

File: src/applause-button.test.js (hunk 3 above)
Comment (chriswilty) [reply src/applause-button.test.js:116]: Hah yes, good question! The test runs were way too close to timing out for comfort, so I reduced these pauses a little; in a subsequent commit (PR to follow) I tackle this in a better way by increasing the overall test timeout.

I did attempt to `useFakeTimers` as we are using Jest under the bonnet, however, that did not appear to work with the jest-puppeteer preset, so it seems we are stuck with _real_ timeouts, hence why these tests take a while to run.

A better solution might be to run most of our tests using Jest and Testing Library, and simply run a couple of smoke tests using puppeteer. Or, we could switch to [Playwright and its Clock API](https://playwright.dev/docs/clock).

File: src/applause-button.js (hunk 2 above)
Comment (chriswilty) [reply src/applause-button.js:52]: Yes. Now that pretty much all browsers support Custom Elements that _extend_ native elements, we no longer need this old-style workaround with its constructor hack.

See the repo for info:
https://github.com/WebReflection/document-register-element?tab=readme-ov-file#deprecated

**Pull request merged**